lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -351,6 +351,15 @@ defmodule RealtimeWeb.RealtimeChannel do
     {:noreply, socket}
   end
 
+  def handle_in(
+        "access_token",
+        %{"access_token" => refresh_token},
+        %{assigns: %{access_token: _access_token}} = socket
+      )
+      when is_nil(refresh_token) do
+    {:noreply, socket}
```

Comment:
@chasers i highly recommend that we shut down the channel at this point like we did [below](https://github.com/supabase/realtime/pull/638/files#diff-2a3a48674821fef237a3451af5d33ca977ee5c7d2f6bc8ca18bd7f18b44a7a26R399). there shouldn't be any reason why any client lib should be sending updated nil access tokens but if they do we should err on the side of caution.

Reply:
Pretty sure that would break everybody. @dshukertjr 's [example app](https://github.com/dshukertjr/realtime-debug/blob/main/src/pages/index.tsx) was sending nulls.

Can you check the js lib?

Reply:
https://github.com/supabase/realtime-js/blob/c9dc9c953cb6bb3e9440749d393e944b63a0a0f9/src/RealtimeClient.ts#L273

Reply:
I think I found why `accessToken` is null here. I opened a PR to fix it!
https://github.com/supabase/realtime-js/pull/245

Reply:
@dshukertjr I assume you’re going to check the dart lib too?

Reply:
@chasers Yup! Once the PR on the JS library is merged, I will add the same implementation on the Dart library as well!

Reply:
@dshukertjr good find and appreciate the PR!